artifacts.go

```diff
@@ -32,6 +32,39 @@ import (
 // Version of the mender-artifact CLI tool
 var Version = "unknown"
 
+func version(c *cli.Context) int {
+	version := c.Int("version")
+	// set version to 2 if artifact is signed and version
+	// is not set explicitly
+	if !c.IsSet("version") && len(c.String("key")) != 0 {
+		version = 2
+	}
+	return version
+}
+
+func artifactWriter(f *os.File, c *cli.Context) (*awriter.Writer, error) {
+	if len(c.String("key")) != 0 {
+		privateKey, err := getKey(c.String("key"))
+		if err != nil {
+			return nil, err
+		}
+		return awriter.NewWriterSigned(f, artifact.NewSigner(privateKey)), nil
+	}
+	return awriter.NewWriter(f), nil
+}
+
+func scripts(c *cli.Context) (*artifact.Scripts, error) {
+	scr := artifact.Scripts{}
+	if len(c.StringSlice("script")) > 0 {
+		for _, script := range c.StringSlice("script") {
```

Comment:
You might be able to lose the if here. If the range is zero it should just skip over the loop.